mlir/include/mlir/Analysis/SliceAnalysis.h

```diff
@@ -138,13 +138,17 @@ void getForwardSlice(Value root, SetVector<Operation *> *forwardSlice,
 /// Assuming all local orders match the numbering order:
 ///    {1, 2, 5, 3, 4, 6}
 ///
-void getBackwardSlice(Operation *op, SetVector<Operation *> *backwardSlice,
-                      const BackwardSliceOptions &options = {});
+/// This function returns whether the backwards slice was able to be
+/// successfully computed, and failure if it was unable to determine the slice.
```

Comment:
That isn't a helpful comment IMO: just seeing the LogicalResult I understand that "this can fail".

What I'd expect is describe the failure **condition** here. What is a prerequisite  of the API.

Reply:
Ping on this?

Reply:
ah sorry for delay, fell of my stack. addressed here: https://github.com/llvm/llvm-project/pull/142223